va/va.go

```diff
@@ -327,15 +372,35 @@ func (va *ValidationAuthorityImpl) validateTLSSNI01WithZName(ctx context.Context
 		},
 	}
 	if problem != nil {
-		return validationRecords, problem
+		return nil, validationRecords, problem
 	}
+	thisRecord := &validationRecords[0]
 
 	// Make a connection with SNI = nonceName
 	portString := strconv.Itoa(va.tlsPort)
 	hostPort := net.JoinHostPort(addr.String(), portString)
-	validationRecords[0].Port = portString
+	thisRecord.Port = portString
 
 	certs, problem := va.getTLSSNICerts(hostPort, identifier, challenge, zName)
+
+	// If there was a problem getting the certificates and the IPv6 first feature
+	// is enabled, then try to fetch the certificates again using a fallback
+	// address
+	if problem != nil && features.Enabled(features.IPv6First) {
```

Comment:
Discussed offline. Conclusion: moving this fallback code into getTLSSNICerts would mean adding a ValidationRecord parameter and deepening the pattern where we [treat ValidationRecords as mutable input/output](https://github.com/letsencrypt/boulder/issues/2730). We can go ahead with this approach, and fix it up in https://github.com/letsencrypt/boulder/issues/2730.